lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoders.java

```diff
@@ -469,6 +469,32 @@ public void skip(int n)
         };
     }
 
+    public ValueDecoder<int[]> getInt96ToLongTimestampWithTimeZoneDecoder(ParquetEncoding encoding)
+    {
+        checkArgument(
+                field.getType() instanceof TimestampWithTimeZoneType timestampType && !timestampType.isShort(),
+                "Trino type %s is not a long timestamp",
+                field.getType());
+        int precision = ((TimestampWithTimeZoneType) field.getType()).getPrecision();
+        return new InlineTransformDecoder<>(
+                getInt96TimestampDecoder(encoding),
+                (values, offset, length) -> {
+                    for (int i = offset; i < offset + length; i++) {
+                        long epochSeconds = decodeFixed12First(values, i);
+                        int nanosOfSecond = decodeFixed12Second(values, i);
+                        if (precision < 9) {
+                            nanosOfSecond = (int) round(nanosOfSecond, 9 - precision);
+                        }
+                        long utcMillis = epochSeconds * MILLISECONDS_PER_SECOND + floorDiv(nanosOfSecond, NANOSECONDS_PER_MILLISECOND);
+                        encodeFixed12(
+                                packDateTimeWithZone(utcMillis, UTC_KEY),
+                                floorMod(nanosOfSecond, NANOSECONDS_PER_MILLISECOND) * PICOSECONDS_PER_NANOSECOND,
```

Comment:
Why is this different from 

https://github.com/trinodb/trino/blob/eda8fdfef81e83bb81c09f3c99d76c75a70e6558/lib/trino-parquet/src/main/java/io/trino/parquet/reader/decoders/ValueDecoders.java#L429

Reply:
It's the same thing either way, we can skip floorDiv/floorMod and use division and modulo directly as `nanosOfSecond` is guaranteed to be positive by this point